com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassPathTracing.hlsl

```diff
@@ -281,7 +288,18 @@ void ComputeSurfaceScattering(inout PathIntersection pathIntersection : SV_RayPa
     }
     #endif
 
+    AOVData aovData;
+    {
+        aovData.albedo = surfaceData.color;
+        aovData.normal = fragInput.tangentToWorld[2];
+    }
+
 #endif // SHADER_UNLIT
+
```

Comment:
It looks like the shader AOV code is called no matter what (or is the compiler smart enough to prune it when not used?).

Reply:
I don't think that we can rely on the compiler to strip this out. 

As far as I can see, avoiding this would require to introduce a new global material variant "#pragma shader_feature ENABLE_AOV" that we could enable programmatically with Shader.EnableKeyword. However this would double the shader variants, so I'm not sure if this is a good option. 

Keep in mind that part of this computation (certainly the motion vectors) can be moved to the ray-gen shader after we remove the recursion, and in this case we can easily strip-it out. Right now I explicitly avoid the AOV writes in the ray-gen when AOV is not active, but not the computations in the ray tracing shader pass.

So, should we revisit this after the recursion removal or do you think we should do something now?

Reply:
Yes fair enough, let's address that later if still needed.